Common/Securities/CashBuyingPowerModel.cs

```diff
@@ -178,7 +178,7 @@ public GetMaximumOrderQuantityForTargetValueResult GetMaximumOrderQuantityForTar
             var unitPrice = new MarketOrder(security.Symbol, 1, DateTime.UtcNow).GetValue(security);
             if (unitPrice == 0)
             {
-                return new GetMaximumOrderQuantityForTargetValueResult(0, $"The price of the {security.Symbol.Value} security is zero because it does not have any market data yet. When the security price is set this security will be ready for trading.");
+                return new GetMaximumOrderQuantityForTargetValueResult(0, $"The internal cash feed required for converting {security.QuoteCurrency.Symbol} to {CashBook.AccountCurrency} does not have any data yet (or market may be closed).");
```

Comment:
Maybe it would be better to explicitly check that the quote currency has a non-zero conversion rate for this message. Looking at the code now, I could see 2 reasons this error would be raised, if the security price is zero or if the conversion rate security price is zero, or did I miss something?

I think this questions are some incentive to put these error messages under unit test.

Reply:
This method assumes the security price is not zero because this check is already done earlier (in `CalculateOrderQuantity`), this method will never be called if `security.Price == 0`.

Reply:
Very well, although I do want to caution us from thinking something *will never be called* due to the way our code is *currently* set up. It's important to view software components in isolation as it reduces logical coupling between components. For example, the engine doesn't currently call this method if `securityPrice == 0` but that doesn't mean a user can't call this method directly and/or a future refactoring could change the current call patterns.

I'm a fan of not *implementing* the same checks twice, but it is often necessary to *perform* the same checks twice. Ideally they're implemented in a single place and reused everywhere that requires those checks.

@StefanoRaggi -- I defer to your judgement regarding this comment thread, but would appreciate the addition of unit tests so that the conditions surrounding each message are well documented and insulated from potential regressions.